stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ActivityTracker.java

```
@@ -0,0 +1,209 @@
+          AutomaticTracker.newInstance(application, this /* tracker */);
+      if (automaticTracker != null) {
+        automaticTracker.register();
+        mAutomaticTracker = automaticTracker;
+        return true;
+      }
+    }
+    return false;
+  }
+
+  public boolean endTracking() {
+    if (mAutomaticTracker != null) {
+      mAutomaticTracker.unregister();
+      return true;
+    }
+    return false;
+  }
+
+  public void add(Activity activity) {
+    Util.throwIfNull(activity);
+    Util.throwIfNot(Looper.myLooper() == Looper.getMainLooper());
+    mActivities.add(activity);
+    mObservable.notifyActivityAdded(activity);
+  }
+
+  public void remove(Activity activity) {
+    Util.throwIfNull(activity);
+    Util.throwIfNot(Looper.myLooper() == Looper.getMainLooper());
+    if (mActivities.remove(activity)) {
+      mObservable.notifyActivityRemoved(activity);
+    }
+  }
+
+  public List<Activity> getActivitiesView() {
+    return mActivitiesUnmodifiable;
+  }
+
+  private static class ActivityTrackerObservable extends Observable<Listener> {
+    public void notifyActivityAdded(Activity activity) {
+      synchronized (mObservers) {
+        int N = mObservers.size();
+        for (int i = 0; i < N; i++) {
+          Listener listener = mObservers.get(i);
+          listener.onActivityAdded(activity);
+        }
+      }
+    }
+
+    public void notifyActivityRemoved(Activity activity) {
+      synchronized (mObservers) {
+        int N = mObservers.size();
+        for (int i = 0; i < N; i++) {
+          Listener listener = mObservers.get(i);
+          listener.onActivityRemoved(activity);
+        }
+      }
+    }
+  }
+
+  public interface Listener {
+    public void onActivityAdded(Activity activity);
+    public void onActivityRemoved(Activity activity);
+  }
+
+  private static abstract class AutomaticTracker {
+    @Nullable
+    public static AutomaticTracker newInstance(Application application, ActivityTracker tracker) {
```

If this returns null, I'd prefer a name which makes that more apparent, e.g. tryNewInstance(). This makes is much more obvious to the caller that it might return null, and that its behavior doesn't just match the 'new' operator (which throws instead of returning null). It certainly makes it easier when reviewing code that calls a method like this.